kong/db/schema/init.lua

```diff
@@ -1805,8 +1805,12 @@ function Schema:process_auto_fields(data, context, nulls, opts)
         end
       end
 
-      if is_select and sdata.translate_backwards and not(opts and opts.hide_shorthands) then
-        data[sname] = table_path(data, sdata.translate_backwards)
+      if is_select and sdata.deprecation and sdata.deprecation.replaced_with and #sdata.deprecation.replaced_with > 0 and not(opts and opts.hide_shorthands) then
+        if sdata.deprecation.replaced_with[1].reverse_mapping_function then
+          data[sname] = sdata.deprecation.replaced_with[1].reverse_mapping_function(data)
+        else
+          data[sname] = table_path(data, sdata.deprecation.replaced_with[1].path)
+        end
       end
     end
```

Comment:
There is a lot going on there. Perhaps adding variable makes it a bit more easier to parse:

```suggestion
      if is_select and not (opts and opts.hide_shorthands) then
        local replaced_with = sdata.deprecation and sdata.deprecation.replaced_with
                                                and sdata.deprecation.replaced_with[1]
        if replaced_with then
          if replaced_with.reverse_mapping_function then
            data[sname] = replaced_with.reverse_mapping_function(data)
          else
            data[sname] = table_path(data, replaced_with.path)
          end
        end
      end
```

Not sure, just an idea.

Reply:
Done ✅